### ISSUE TITLE
Cargo book should explain that features must be additive

### ISSUE THREAD

**Issue by jimblandy** [rust-lang/cargo#8458]
Cargo features must be additive, because if two crates agree on the version of a common dependency, but want different features from it, that dependency is built only once, with the union of the features requested. So if a crate can meet a dependent's expectations with a given set of features enabled, it must still meet those expectations if additional features are enabled.

This is a pretty fundamental restriction on how you can use features, but the Cargo book doesn't mention it in the section on features. It would be a natural fit in the 'Rules' section.

**Comment by Eh2406** [comment]
That is an important restriction on features, Yes. If there is a good place to add it to docs I'd be open to seeing a PR. 

Thank you!

**Comment by ehuss** [comment]
Thanks for the report!  This is a duplicate of #4328, so closing in favor of that.

**Comment by jimblandy** [comment]
@ehuss D'oh, thanks. I did search, but I guess I didn't find the magic set of terms...
